sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/BuilderHelper.java

```diff
@@ -182,4 +195,22 @@ static void validateCredentials(AzureNamedKeyCredential azureNamedKeyCredential,
                     + usedCredentialsStringBuilder + "."));
         }
     }
+
+    /**
+     * Validates that the client is properly configured to use https.
+     *
+     * @param objectToCheck The object to check for.
+     * @param objectName The name of the object.
+     * @param endpoint The endpoint for the client.
+     */
+    public static void httpsValidation(Object objectToCheck, String objectName, String endpoint, ClientLogger logger) {
+        try {
+            if (objectToCheck != null && !new URI(endpoint).getScheme().equals(StorageConstants.HTTPS)) {
```

Comment:
This doesn't need to be done here as the bearer token credential policy performs this validation:

```java
if ("http".equals(context.getHttpRequest().getUrl().getProtocol())) {
            return Mono.error(new RuntimeException("token credentials require a URL using the HTTPS protocol scheme"));
        }
```